addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/handler/OnkyoHandler.java

```diff
@@ -70,66 +81,51 @@ public OnkyoHandler(Thing thing, UpnpIOService upnpIOService, AudioHTTPServer au
     @Override
     public void initialize() {
         logger.debug("Initializing handler for Onkyo Receiver");
-        if (this.getConfig().get(HOST_PARAMETER) != null) {
-            String host = (String) this.getConfig().get(HOST_PARAMETER);
-            Integer port = 60128;
-            Object portObj = this.getConfig().get(TCP_PORT_PARAMETER);
-            if (portObj != null) {
-                if (portObj instanceof Number) {
-                    port = ((Number) portObj).intValue();
-                } else if (portObj instanceof String) {
-                    port = Integer.parseInt(portObj.toString());
-                }
-            }
+        configuration = getConfigAs(OnkyoDeviceConfiguration.class);
+        logger.info("Using configuration: {}", configuration.toString());
 
-            long refreshInterval;
-            try {
-                refreshInterval = ((BigDecimal) this.getConfig().get(REFRESH_INTERVAL)).longValue();
-            } catch (Exception e) {
-                refreshInterval = 60;
-                logger.warn("No refresh Interval defined using {}s", refreshInterval);
-            }
-            logger.warn("No refresh Interval defined using {}s", refreshInterval);
+        connection = new OnkyoConnection(configuration.ipAddress, configuration.port);
+        connection.addEventListener(this);
 
-            connection = new OnkyoConnection(host, port);
-            connection.addEventListener(this);
+        scheduler.execute(new Runnable() {
 
-            logger.debug("Connected to Onkyo Receiver @{}", connection.getConnectionName());
+            @Override
+            public void run() {
+                logger.debug("Open connection to Onkyo Receiver @{}", connection.getConnectionName());
+                connection.openConnection();
+                if (connection.isConnected()) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }
+        });
+
+        if (configuration.refreshInterval > 0) {
+            // Start resource refresh updater
+            resourceUpdaterFuture = scheduler.scheduleAtFixedRate(new Runnable() {
```

Comment:
As per the binding design guidelines _For periodically executed jobs that do not require a fixed rate scheduleWithFixedDelay should be preferred over scheduleAtFixedRate._